tests/scenario/test_machine_charm/test_cos_agent_e2e.py

```diff
@@ -94,50 +103,119 @@ def __init__(self, framework: Framework):
                 refresh_events=[self.on.cos_agent_relation_changed],
             )
 
+    # Step 1: gagent is deployed and ends in "unknown" state
+    # since it is not related to the principal charm.
     cos_agent = Relation("cos-agent")
-
-    state_out = State(
-        relations=[
-            cos_agent,
-        ]
-    ).trigger(
-        cos_agent.changed_event(remote_unit_id=1),
+    ctx = Context(
         charm_type=MyPrincipal,
         meta=MyPrincipal.META,
         charm_root=vroot,
     )
 
-    # step 2: gagent is notified that the principal has touched its relation data
+    state = State(relations=[cos_agent])
+    state_out = ctx.run(cos_agent.changed_event(remote_unit_id=1), state=state)
+    assert state_out.status.unit.name == "unknown"
+
+    # Step 2: gagent is related to principal charm and ends in "blocked" status
+    # since there are missing relations:
+    #  - send-remote-write
+    #  - logging-consumer
+    #  - grafana-dashboards-provider
     peer = PeerRelation("peers")
     cos_agent1 = SubordinateRelation(
         "cos-agent",
         primary_app_name="mock-principal",
```

Comment:
**change** if you upgrade to scenario 3.0.2 you'll see that `primary-app-name` becomes `remote-app-name`
also if it's not pinned already, please pin the scenario version in tox/pyproject.toml

Reply:
I'm migrating to 3.x.. and if it's in 3.0.2 I'm ok with using `remote-app-name`... WDYT?